src/models/collaboration/usePatchCollaboration.js

```diff
@@ -1,48 +1,76 @@
-import { useState } from 'react';
 import axios from 'axios';
-import { get } from 'lodash-es';
-import { formatError } from '../../utils/formatters';
+import { useMutation } from 'react-query';
 
 export default function usePatchCollaboration() {
-  const [loading, setLoading] = useState(false);
-  const [error, setError] = useState(null);
-  const [success, setSuccess] = useState(false);
+  const collabPatchMutation = useMutation(async dataObj => {
+    const collabId = dataObj?.collabId;
 
-  const patchCollaboration = async (collaborationId, data) => {
-    try {
-      setLoading(true);
-      const patchResponse = await axios({
-        url: `${__houston_url__}/api/v1/collaborations/${collaborationId}`,
-        withCredentials: true,
-        method: 'patch',
-        data,
-      });
-      const responseStatus = get(patchResponse, 'status');
-      const successful = responseStatus === 200;
-      if (successful) {
-        setLoading(false);
-        setSuccess(true);
-        setError(null);
-        return true;
-      }
-
-      setError(formatError(patchResponse));
-      setSuccess(false);
-      return false;
-    } catch (postError) {
-      setLoading(false);
-      setError(formatError(postError));
-      setSuccess(false);
-      return false;
+    if (dataObj?.dataReverse) {
+      // it's two-way
+      return Promise.all([
+        axios.request({
+          url: `${__houston_url__}/api/v1/collaborations/${collabId}`,
+          withCredentials: true,
+          method: 'patch',
+          data: dataObj?.dataForward,
+        }),
+        axios.request({
+          url: `${__houston_url__}/api/v1/collaborations/${collabId}`,
+          withCredentials: true,
+          method: 'patch',
+          data: dataObj?.dataReverse,
+        }),
+      ]);
     }
+    return axios({
+      url: `${__houston_url__}/api/v1/collaborations/${collabId}`,
+      withCredentials: true,
+      method: 'patch',
+      data: dataObj?.dataForward,
+    });
+  });
+
+  const patchCollaboration = (
+    collabId,
+    dataForward,
+    dataReverse = null,
+  ) => {
+    let dataObj = {};
+    dataObj.collabId = collabId;
+    dataObj.dataForward = dataForward;
+    if (dataReverse) dataObj.dataReverse = dataReverse;
+    return collabPatchMutation.mutate(dataObj);
+  };
+
+  const pathCollaborationAsync = (
+    collabId,
+    dataForward,
+    dataReverse = null,
+  ) => {
+    let dataObj = {};
+    dataObj.collabId = collabId;
+    dataObj.dataForward = dataForward;
+    if (dataReverse) dataObj.dataReverse = dataReverse;
+    return collabPatchMutation.mutateAsync(dataObj);
   };
 
+  const {
+    isSuccess,
+    isError,
+    isLoading,
+    error: collabPatchError,
+  } = collabPatchMutation;
+
+  const error = collabPatchError
+    ? collabPatchError.toJSON().message
+    : null;
+
   return {
     patchCollaboration,
-    loading,
+    pathCollaborationAsync,
```

Comment:
do we need both? seems to me like async is all we need

Reply:
if we do need both i think you could make the code above a little less repetetive

Reply:
Done